app/models/test.rb

```diff
@@ -5,10 +5,18 @@ class Test < ApplicationRecord
   has_many :user_tests, dependent: :destroy
   has_many :users, through: :user_tests
 
-  def self.get_by_category(category)
-    Test.joins(:category)
-        .where(categories: {title: category})
-        .pluck(:title)  
+  scope :easy, -> { where(level: 0..1) }
+  scope :normal, -> { where(level: 2..4) }
+  scope :hard, -> { where("level >= ?", 5) }
+  scope :by_level, -> (level) { where("level = ?", level) }
```

Comment:
```suggestion
  scope :by_level, -> (level) { where(level: level) }
```
лучше через хеши писать, там, где возможно
короче и легче читать